pkg/operator2/configsync.go

```diff
@@ -91,64 +91,189 @@ type sourceData struct {
+			)
+
+		case configv1.IdentityProviderTypeKeystone:
+			p := pc.Keystone
+
+			ca := p.CA.Name
+			caDest, caData := newSourceDataIDPConfigMap(i, ca, corev1.ServiceAccountRootCAKey)
+
+			clientCert := p.TLSClientCert.Name
+			clientCertDest, clientCertData := newSourceDataIDPSecret(i, clientCert, corev1.TLSCertKey)
+
+			clientKey := p.TLSClientKey.Name
+			clientKeyDest, clienKeyData := newSourceDataIDPSecret(i, clientKey, corev1.TLSCertKey)
 
 			out = append(out,
 				idpSyncData{
+					configMaps: map[string]sourceData{caDest: caData},
 					secrets: map[string]sourceData{
-						dest: {
-							src:    fileData,
-							path:   path,
-							volume: volume,
-							mount:  mount,
-						},
+						clientCertDest: clientCertData,
+						clientKeyDest:  clienKeyData,
 					},
 				},
 			)
+
+		case configv1.IdentityProviderTypeLDAP:
+			p := pc.LDAP
+
+			ca := p.CA.Name
+			caDest, caData := newSourceDataIDPConfigMap(i, ca, corev1.ServiceAccountRootCAKey)
+
+			bindPassword := p.BindPassword.Name
+			bindPasswordDest, bindPasswordData := newSourceDataIDPSecret(i, bindPassword, configv1.BindPasswordKey)
+
+			out = append(out,
+				idpSyncData{
+					configMaps: map[string]sourceData{caDest: caData},
+					secrets:    map[string]sourceData{bindPasswordDest: bindPasswordData},
+				},
+			)
+
 		case configv1.IdentityProviderTypeOpenID:
 			p := pc.OpenID
 
 			ca := p.CA.Name
-			caDest := getIDPName(i, ca, corev1.ServiceAccountRootCAKey)
-			caVolume, caMount, caPath := configMapVolume(i, caDest, corev1.ServiceAccountRootCAKey)
+			caDest, caData := newSourceDataIDPConfigMap(i, ca, corev1.ServiceAccountRootCAKey)
 
 			clientSecret := p.ClientSecret.Name
-			clientSecretDest := getIDPName(i, clientSecret, configv1.ClientSecretKey)
-			clientSecretVolume, clientSecretMount, clientSecretPath := secretVolume(i, clientSecretDest, configv1.ClientSecretKey)
+			clientSecretDest, clientSecretData := newSourceDataIDPSecret(i, clientSecret, configv1.ClientSecretKey)
 
 			out = append(out,
 				idpSyncData{
-					configMaps: map[string]sourceData{
-						caDest: {
-							src:    ca,
-							path:   caPath,
-							volume: caVolume,
-							mount:  caMount,
-						},
-					},
+					configMaps: map[string]sourceData{caDest: caData},
 					secrets: map[string]sourceData{
-						clientSecretDest: {
-							src:    clientSecret,
-							path:   clientSecretPath,
-							volume: clientSecretVolume,
-							mount:  clientSecretMount,
-						},
+						clientSecretDest: clientSecretData,
 					},
 				},
 			)
+
+		case configv1.IdentityProviderTypeRequestHeader:
+			p := pc.RequestHeader
+
+			clientCA := p.ClientCA.Name
+			clientCADest, clientCAData := newSourceDataIDPConfigMap(i, clientCA, corev1.ServiceAccountRootCAKey)
+
+			out = append(out,
+				idpSyncData{
+					configMaps: map[string]sourceData{clientCADest: clientCAData},
+				},
+			)
+
 		default:
-			panic("TODO")
+			return nil // TODO: some erroring
```

Comment:
`return fmt.Errorf(“unknown IdentityProvider type: %s”, pc.Type)`

Reply:
And thank you, no panics!

Reply:
The function currently only returns `[]idpSyncData`, I'd maybe fix this when gets merged with `convertProviderConfigToOsinBytes()`

Reply:
icky, ok. for another pr